lib/extract-numbers.js

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2015, Yahoo Inc.
+ * Copyrights licensed under the New BSD License.
+ * See the accompanying LICENSE file for terms.
+ */
+'use strict';
+
+var p = require('path');
+
+var getParentLocale = require('./locales').getParentLocale;
+var hasNumberFields   = require('./locales').hasNumberFields;
+var normalizeLocale = require('./locales').normalizeLocale;
+
+// The set of CLDR date field names that are used in FormatJS.
+var NUMBER_FIELD_NAMES = [
+    'decimalFormats-numberSystem-latn',
+    'currencyFormats-numberSystem-latn',
+];
+
+module.exports = function extractNumberFields(locales) {
+    // The CLDR states that the "root" locale's data should be used to fill in
+    // any missing data as its data is the default.
+    var defaultFields = loadNumberFields('root');
+
+    var fields = {};
+    var hashes = {};
+
+    // Loads and caches the relative fields for a given `locale` because loading
+    // and transforming the data is expensive.
+    function getNumberFields(locale) {
+        var relativeFields = fields[locale];
+        if (relativeFields) {
+            return relativeFields;
+        }
+
+        if (hasNumberFields(locale)) {
+            relativeFields = fields[locale] = loadNumberFields(locale);
+            return relativeFields;
+        }
+    }
+
+    // Hashes and caches the `fields` for a given `locale` to avoid hashing more
+    // than once since it could be expensive.
+    function hashFields(locale, fields) {
+        var hash = hashes[locale];
+        if (hash) {
+            return hash;
+        }
+
+        hash = hashes[locale] = JSON.stringify(fields);
+        return hash;
+    }
+
+    // We want to de-dup data that can be referenced from upstream in the
+    // `locale`'s hierarchy when that locale's relative fields are the _exact_
+    // same as one of its ancestors. This will traverse the hierarchy for the
+    // given `locale` until it finds an ancestor with same same relative fields.
+    // When an ancestor can't be found, a data entry must be created for the
+    // `locale` since its relative fields are unique.
+    function findGreatestAncestor(locale) {
```

Comment:
@jasonmit I'm not sure what you think of this, but there is some common logic with extract-relative.  Wasn't sure how much surface area to touch as we are a fork and not sure of future plans.  

I'm ok with leaving it for now as well.

Reply:
I'm fine with duplication here.  If we find ourselves repeating this more we can abstract it away.